pkg/controller/node/nodecontroller.go

```diff
@@ -489,6 +490,9 @@ func (nc *NodeController) monitorNodeStatus() error {
 			// Report node event.
 			if readyCondition.Status != api.ConditionTrue && lastReadyCondition.Status == api.ConditionTrue {
 				nc.recordNodeStatusChange(node, "NodeNotReady")
+				if err = nc.markAllPodsNotReady(node.Name); err != nil {
```

Comment:
Not sure why this method is in a poll loop instead of triggered by a watch (not your problem, just saying), there's a chance it will still not realize as soon as a node falls off the netowork so the reboot test (the one that needs the pods to become not ready) can still fail.  How often does this poll loop run?